[BUG] Alluxio config `pathsToReplace` does not overwrite `automount` config.

**Describe the bug**
Alluxio config `pathsToReplace` does not overwrite `automount` config.
From the doc, we see:
```
spark.rapids.alluxio.pathsToReplace
If you set this config, spark.rapids.alluxio.automount.enabled won't be valid.
```
If we set both of them, then got error, see the reproduce steps.

**Steps/Code to reproduce bug**
Test case:
Set `automount.enabled` and `pathsToReplace` at the same time:
``` python
def test_my1():
    with_gpu_session(lambda spark: spark.read.parquet("s3a://liangcail/chongg-test").count(),
                     {
                         "spark.rapids.alluxio.automount.enabled": "true",
                         "spark.rapids.alluxio.pathsToReplace": "s3a://nds20->alluxio://10.59.239.52:19998/nds20"
                     })
```
                     
error:
```
E                   py4j.protocol.Py4JJavaError: An error occurred while calling o310.count.
E                   : java.util.NoSuchElementException: None.get
E                   	at scala.None$.get(Option.scala:529)
E                   	at scala.None$.get(Option.scala:527)
E                   	at com.nvidia.spark.rapids.AlluxioUtils$.$anonfun$autoMountIfNeeded$3(AlluxioUtils.scala:459)
E                   	at scala.collection.immutable.List.flatMap(List.scala:366)
E                   	at com.nvidia.spark.rapids.AlluxioUtils$.$anonfun$autoMountIfNeeded$1(AlluxioUtils.scala:458)
E                   	at scala.collection.immutable.List.flatMap(List.scala:366)
E                   	at com.nvidia.spark.rapids.AlluxioUtils$.autoMountIfNeeded(AlluxioUtils.scala:457)
E                   	at org.apache.spark.sql.rapids.GpuFileSourceScanExec.selectedPartitions$lzycompute(GpuFileSourceScanExec.scala:126)
E                   	at org.apache.spark.sql.rapids.GpuFileSourceScanExec.selectedPartitions(GpuFileSourceScanExec.scala:118)
E                   	at org.apache.spark.sql.rapids.GpuFileSourceScanExec.dynamicallySelectedPartitions$lzycompute(GpuFileSourceScanExec.scala:166)
```

code
`isAlluxioAutoMountTaskTime` and `isAlluxioPathsToReplaceTaskTime` are both true.
[Link](https://github.com/NVIDIA/spark-rapids/blob/branch-22.12/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala#L123-L133)

``` scala
    if (AlluxioCfgUtils.isAlluxioAutoMountTaskTime(rapidsConf, relation.fileFormat)) {
      alluxioPathReplacementMap = AlluxioUtils.autoMountIfNeeded(rapidsConf, pds,
        relation.sparkSession.sparkContext.hadoopConfiguration,
        relation.sparkSession.conf)
    } else if (AlluxioCfgUtils.isAlluxioPathsToReplaceTaskTime(rapidsConf, relation.fileFormat)) {
      // this is not ideal, here we check to see if we will replace any paths, which is an
      // extra iteration through paths
      alluxioPathReplacementMap = AlluxioUtils.checkIfNeedsReplaced(rapidsConf, pds,
        relation.sparkSession.sparkContext.hadoopConfiguration,
        relation.sparkSession.conf)
    }
```
**Expected behavior**
Should first check `isAlluxioPathsToReplaceTaskTime` and then `isAlluxioAutoMountTaskTime`.
``` scala
    if (AlluxioCfgUtils.isAlluxioAutoMountTaskTime(rapidsConf, relation.fileFormat)) {
    } else if (AlluxioCfgUtils.isAlluxioPathsToReplaceTaskTime(rapidsConf, relation.fileFormat)) {
    }
```

====>>>>

``` scala

    if (AlluxioCfgUtils.isAlluxioPathsToReplaceTaskTime(rapidsConf, relation.fileFormat)) {
    } else if (AlluxioCfgUtils.isAlluxioAutoMountTaskTime(rapidsConf, relation.fileFormat)) {
    }
```

**Environment details (please complete the following information)**
 - Environment location: databricks.

--- a/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala
+++ b/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala
@@ -120,17 +120,18 @@ case class GpuFileSourceScanExec(
     val startTime = System.nanoTime()
     val pds = relation.location.listFiles(
         partitionFilters.filterNot(isDynamicPruningFilter), dataFilters)
-    if (AlluxioCfgUtils.isAlluxioAutoMountTaskTime(rapidsConf, relation.fileFormat)) {
-      alluxioPathReplacementMap = AlluxioUtils.autoMountIfNeeded(rapidsConf, pds,
-        relation.sparkSession.sparkContext.hadoopConfiguration,
-        relation.sparkSession.conf)
-    } else if (AlluxioCfgUtils.isAlluxioPathsToReplaceTaskTime(rapidsConf, relation.fileFormat)) {
+    if (AlluxioCfgUtils.isAlluxioPathsToReplaceTaskTime(rapidsConf, relation.fileFormat)) {
       // this is not ideal, here we check to see if we will replace any paths, which is an
       // extra iteration through paths
       alluxioPathReplacementMap = AlluxioUtils.checkIfNeedsReplaced(rapidsConf, pds,
         relation.sparkSession.sparkContext.hadoopConfiguration,
         relation.sparkSession.conf)
+    } else if (AlluxioCfgUtils.isAlluxioAutoMountTaskTime(rapidsConf, relation.fileFormat)) {
+      alluxioPathReplacementMap = AlluxioUtils.autoMountIfNeeded(rapidsConf, pds,
+        relation.sparkSession.sparkContext.hadoopConfiguration,
+        relation.sparkSession.conf)
     }
+
     logDebug(s"File listing and possibly replace with Alluxio path " +
       s"took: ${System.nanoTime() - startTime}")
 

